onnx/test/helper_test.py

```diff
@@ -724,6 +724,44 @@ def test_make_tensor_raw(tensor_dtype: int) -> None:
     np.testing.assert_equal(np_array, numpy_helper.to_array(tensor))
 
 
+class TestHelperMappingFunctions(unittest.TestCase):
+    # TODO (#4554): remove these tests about catching warnings after the deprecation period
+    # Test these new functions should not raise any deprecation warnings
+    @pytest.mark.filterwarnings("error::DeprecationWarning")
+    def test_tensor_dtype_to_np_dtype_not_throw_warning(self) -> None:
+        _ = helper.tensor_dtype_to_np_dtype(TensorProto.FLOAT)
+
+    @pytest.mark.filterwarnings("error::DeprecationWarning")
+    def test_tensor_dtype_to_storage_tensor_dtype_not_throw_warning(self) -> None:
+        _ = helper.tensor_dtype_to_storage_tensor_dtype(TensorProto.FLOAT)
+
+    @pytest.mark.filterwarnings("error::DeprecationWarning")
+    def test_tensor_dtype_to_field_not_throw_warning(self) -> None:
+        _ = helper.tensor_dtype_to_field(TensorProto.FLOAT)
+
+    @pytest.mark.filterwarnings("error::DeprecationWarning")
+    def test_np_dtype_to_tensor_dtype_not_throw_warning(self) -> None:
+        _ = helper.np_dtype_to_tensor_dtype(np.dtype("float32"))
+
+    def test_tensor_dtype_to_np_dtype_bfloat16(self) -> None:
+        self.assertEqual(
+            helper.tensor_dtype_to_np_dtype(TensorProto.BFLOAT16), np.dtype("float32")
+        )
+
+    def test_tensor_dtype_to_storage_tensor_dtype_bfloat16(self) -> None:
+        self.assertEqual(
+            helper.tensor_dtype_to_storage_tensor_dtype(TensorProto.BFLOAT16),
+            TensorProto.UINT16,
+        )
+
+    # BFloat16 tensor uses TensorProto.UINT16 as storage type;
+    # And the field name for TensorProto.UINT16 is int32_data
+    def test_tensor_dtype_to_field_bfloat16(self) -> None:
+        self.assertEqual(
+            helper.tensor_dtype_to_field(TensorProto.BFLOAT16), "int32_data"
+        )
+
+
```

Comment:
I can be done in another PR but I would check all the types:

```python
    def test_numeric_types(self):  # type: ignore
        dtypes = [
            np.float16,
            np.float32,
            np.float64,
            np.int8,
            np.int16,
            np.int32,
            np.int64,
            np.uint8,
            np.uint16,
            np.uint32,
            np.uint64,
            np.complex64,
            np.complex128,
        ]
        for dt in dtypes:
            with self.subTest(dtype=dt):
                t = np.array([0, 1, 2], dtype=dt)
                ot = from_array(t)
                u = to_array(ot)
                self.assertEqual(t.dtype, u.dtype)
                assert_almost_equal(t, u)

    def test_make_tensor(self):  # type: ignore
        for pt, dt in TENSOR_TYPE_TO_NP_TYPE.items():
            if pt == TensorProto.BFLOAT16:
                continue
            with self.subTest(dt=dt, pt=pt, raw=False):
                if pt == TensorProto.STRING:
                    t = np.array([["i0", "i1", "i2"], ["i6", "i7", "i8"]], dtype=dt)
                else:
                    t = np.array([[0, 1, 2], [6, 7, 8]], dtype=dt)
                ot = make_tensor("test", pt, t.shape, t, raw=False)
                self.assertFalse(ot is None)
                u = to_array(ot)
                self.assertEqual(t.dtype, u.dtype)
                self.assertEqual(t.tolist(), u.tolist())
            with self.subTest(dt=dt, pt=pt, raw=True):
                t = np.array([[0, 1, 2], [6, 7, 8]], dtype=dt)
                if pt == TensorProto.STRING:
                    with self.assertRaises(TypeError):
                        make_tensor("test", pt, t.shape, t.tobytes(), raw=True)
                else:
                    ot = make_tensor("test", pt, t.shape, t.tobytes(), raw=True)
                    self.assertFalse(ot is None)
                    u = to_array(ot)
                    self.assertEqual(t.dtype, u.dtype)
                    assert_almost_equal(t, u)
```

Reply:
Thank you @xadupre for the suggestion. It looks good to me, but I am inclined to have it in another PR (this PR more focuses on bug fixes) Feel free to let me know if you still have other concern.